### PR TITLE
Updated MongoDB component definition example

### PR DESCRIPTION
**Pull request by iMichaela**
# Committer Notes

Updated MongoDB component definition to add SC-8 and to set parameters

### All Submissions:

- [x] Have you followed the guidelines in our [Contributing](https://github.com/usnistgov/oscal-content/blob/master/CONTRIBUTING.md) document?
- [x] Have you checked to ensure there aren't other open [Pull Requests](https://github.com/usnistgov/oscal-content/pulls) for the same update/change?
- [x] Have you squashed any non-relevant commits and commit messages? \[[instructions](https://git-scm.com/book/en/v2/Git-Tools-Rewriting-History)\]
- [x] Do all automated CI/CD checks pass?

### Changes to Core Features:

- ~Have you added an explanation of what your changes do and why you'd like us to include them?~ Bug fix, no detailed explanation needed.
- ~Have you written new tests for your core changes, as applicable?~ N/A
- ~Have you included examples of how to use your new feature(s)?~ N/A

### CODE REVIEW

File: src/examples/component-definition/xml/example-component-definition.xml
Comment (david-waltermire-nist) [inline src/examples/component-definition/xml/example-component-definition.xml:58]: This description doesn't add any information. You should consider revising it to containing some additional information. Maybe an explanation that the control implementation is addressed at the statement level and there are no additional details at the control level?

File: src/examples/component-definition/xml/example-component-definition.xml
Comment (david-waltermire-nist) [inline src/examples/component-definition/xml/example-component-definition.xml:68]:
```suggestion
              of transmitted data by encrypting data in transit,
```

**Review by david-waltermire-nist — CHANGES_REQUESTED** [review]
See comments in the component definition file review.

File: src/examples/component-definition/xml/example-component-definition.xml
Comment (iMichaela) [reply src/examples/component-definition/xml/example-component-definition.xml:58]: Done.

File: src/examples/component-definition/xml/example-component-definition.xml
Comment (iMichaela) [reply src/examples/component-definition/xml/example-component-definition.xml]: Done. Thank you.

**Comment by iMichaela** [comment]
@david-waltermire-nist I changed the parameters <insert>  format so the ci/cd pipeline does not chock, but the file has the OSCAL 1.0.0 schema listed and it no longer validates to OSCAL 1.0.0 because of the <insert> format change. What do you advise? I suggest we have an official OSCAL 1.0.1  and make the file the correct version. As it is now, it is not OSCAL 1.0.0 accurate as it claims. If I make it OSCAL 1.0.0 will not pass the ci/cd validation tests.

File: src/examples/component-definition/xml/example-component-definition.xml
Comment (david-waltermire-nist) [inline src/examples/component-definition/xml/example-component-definition.xml:82]: Remove the quotes.

**Comment by iMichaela** [comment]
@david-waltermire-nist - all requested changes were implemented long ago, and the issue described in the message of Sept 23, 2021 was resolved by the oscal 1.0.4. The example validates now and it is ready for approval.

File: src/examples/component-definition/xml/example-component-definition.xml
Comment (aj-stein-nist) [inline src/examples/component-definition/xml/example-component-definition.xml:71]: Minor, non-blocking: One of us had recommended that we move to not manually create newlines in XML `text()` nodes when multi-line like the `description` and we used the Word Wrap feature of our IDEs instead. In the future, it will lead to clean diffing, can we change that before merge?

```suggestion
            <description>
              <p>MongoDB implements TLS 1.x to protect the <insert type="param" id-ref="sc-8_prm_1"/> of transmitted data by encrypting data in transit, preventing unauthorized disclosure or changes to information during transmission.</p>
            </description>
```

File: src/examples/component-definition/xml/example-component-definition.xml
Comment (aj-stein-nist) [inline src/examples/component-definition/xml/example-component-definition.xml:79]: Non-blocking, minor: similar whitespace/wordwrapping recommendation.

```suggestion
          <p>MongoDB implements cryptographic mechanisms (TLS 1.x) to provide cryptographic protection for data in transit.</p>
```

File: src/examples/component-definition/xml/example-component-definition.xml
Comment (aj-stein-nist) [inline src/examples/component-definition/xml/example-component-definition.xml:92]: Non-blocking, minor: similar whitespace/wordwrapping recommendation. Additionally, it might be a nice stylistic approach to wrap the variable name and file path for that variable in `<code/>` blocks for clarity. I leave that up to you. 

```suggestion
              <p>To implement cryptographic mechanisms (aka enable TLS 1.x) to <insert type="param" id-ref="sc-8.1_prm_1"/>, customers need to set the <code>PEMKeyFile</code> option in the configuration  file <code>/etc/mongod.conf</code> to the certificate file's path and restart the component.</p>
```

**Review by aj-stein-nist — APPROVED** [review]
Looks good. I left some minor feedback about how we try to let word-wrapping manage long-lines for cleaner than diffs. If you make those changes with the commit changes button or want me to assist, I can and we can merge.

**Review by aj-stein-nist — APPROVED** [review]

**Review by aj-stein-nist — APPROVED** [review]

**Comment by iMichaela** [comment]
@david-waltermire-nist Is there anything else you think needs to be updated to this PR?

**Comment by aj-stein-nist** [comment]
@iMichaela I am going to re-review this PR and I would like us to sync up next week, adjust if needed, and get ready to merge next week.

**Comment by iMichaela** [comment]
@AJ - This PR should probably be resubmitted against develop after rebased and example updated to OSCAL 1.1.0

**Comment by aj-stein-nist** [comment]
> @aj-stein-nist - This PR should probably be resubmitted against `develop` after rebasing it and updating the example to OSCAL 1.1.0

Sure, or we can rebase this PR on top of that as well. Either way, that is one reason we can circle back next week. 👋

File: src/examples/component-definition/xml/example-component-definition.xml
Comment (aj-stein-nist) [inline src/examples/component-definition/xml/example-component-definition.xml:31]: You do not want to bind the MongoDB organization to the customer role for the component in the CDEF as that customer may not be known. We should stub it without a specific party identified by UUID.

```suggestion
    <responsible-role role-id="customer"/>
```

File: src/examples/component-definition/xml/example-component-definition.xml
Comment (aj-stein-nist) [inline src/examples/component-definition/xml/example-component-definition.xml:86]: You do not want to bind the MongoDB organization to the customer role for the component in the CDEF as that customer may not be known. We should stub it without a specific party identified by UUID.

```suggestion
          <responsible-role role-id="customer"/>
```

**Review by aj-stein-nist — CHANGES_REQUESTED** [review]
I rebased this against the current `develop` branch and it looks good, I noticed two minor issues while reviewing it during validation that are worth correcting. If you are ok with those, we can merge this and then roll it into a release.

File: src/examples/component-definition/xml/example-component-definition.xml
Comment (iMichaela) [reply src/examples/component-definition/xml/example-component-definition.xml:31]: You are right. I did not pay attention. This was done 2 years ago and if I recall, I only added the parent controls. BUT - my fault for not catching it at that time. I agree with eh change.

Comment (aj-stein-nist) [reply src/examples/component-definition/xml/example-component-definition.xml]: No worries . We committed the change in one place not this one as well. I will unresolve this comment and rec, add it in, re-resolve this conversation topic, approve, and merge.

**Review by aj-stein-nist — APPROVED** [review]

File: src/examples/component-definition/xml/example-component-definition.xml
Comment (iMichaela) [reply src/examples/component-definition/xml/example-component-definition.xml]: Thank you. Good to go.

**Pull request merged**
